### PR TITLE
#2109 schema infer max

### PR DESCRIPTION
**Pull request by jychen7**
# Which issue does this PR close?

patch v7.x with #2109 
was merged to `master` at https://github.com/apache/arrow-datafusion/pull/2139

 # Rationale for this change

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

File: datafusion/fuzz-utils/src/lib.rs [hunk 1]
```diff
@@ -26,7 +26,7 @@ pub use env_logger;
 pub fn batches_to_vec(batches: &[RecordBatch]) -> Vec<Option<i32>> {
     batches
         .iter()
-        .map(|batch| {
+        .flat_map(|batch| {
```
Comment (jychen7) [inline datafusion/fuzz-utils/src/lib.rs:29]: same fix as https://github.com/apache/arrow-datafusion/pull/1885/files

**Review by yjshen — APPROVED** [review]
Thanks @jychen7!

**Review by alamb — APPROVED** [review]
Thanks @jychen7

**Comment by alamb** [comment]
Looks like there are some other clippy errors left on the maint branch -- if you want we can merge this PR in as is and then we can fix the clippy errors on the maint branch as a follow on PR

**Comment by jychen7** [comment]
> the clippy error impact quite a few files, would be better we resolve in new PR

https://github.com/apache/arrow-datafusion/pull/2166

**Pull request merged**